packages/backend/src/modules/tracked-txs/modules/l2-costs/L2CostsModule.ts

```diff
@@ -0,0 +1,46 @@
+import { Logger } from '@l2beat/backend-tools'
+import { createPublicClient, http } from 'viem'
+
+import { Config } from '../../../../config'
+import { Peripherals } from '../../../../peripherals/Peripherals'
+import { ViemRpcClient } from '../../../../peripherals/viem-rpc-client/ViemRpcClient'
+import { ApplicationModuleWithUpdater } from '../../../ApplicationModule'
+import { L2CostsUpdater } from './L2CostsUpdater'
+import { L2CostsRepository } from './repositories/L2CostsRepository'
+
+export function createL2CostsModule(
+  config: Config,
+  logger: Logger,
+  peripherals: Peripherals,
+): ApplicationModuleWithUpdater<L2CostsUpdater> | undefined {
+  if (!config.trackedTxsConfig || !config.trackedTxsConfig.uses.l2costs) {
+    logger.info('L2Costs module disabled')
+    return
+  }
+
+  const publicClient = createPublicClient({
+    transport: http(config.trackedTxsConfig.uses.l2costs.providerUrl),
+  })
+  const viemRpcClient = new ViemRpcClient(
```

Comment:
probably after Piotr's refactoring we should use `peripherals.getClient(ViemRpcClient)`

Reply:
[d3d5312](https://github.com/l2beat/l2beat/pull/3007/commits/d3d5312495e73484cbbc35ced72e8b70b8c72679)